bodhi/server/consumers/composer.py

```diff
@@ -184,7 +186,12 @@ def __call__(self, message: fedora_messaging.api.Message):
         message = message.body["msg"]
         resume = message.get('resume', False)
         agent = message.get('agent')
-        notifications.publish(topic="composer.start", msg=dict(agent=agent), force=True)
+        # This callback is run in a thread by Twisted, and must therefore call
+        # the messaging API with the proper Twisted wrapper to ensure thread
+        # safety.
+        blockingCallFromThread(
+            reactor, notifications.publish,
+            topic="composer.start", msg=dict(agent=agent), force=True)
```

Comment:
Would it be cleaner if we made ```notifications.publish()``` do this when passed a new option (perhaps a boolean like ```run_in_reactor``` or some better name)? I don't mind this, but it might be easier to remember to do it if it were done in one place and had an easy parameter to select the behaviour.

What do you think? I'm fine with this style if you prefer it.

Reply:
I tried that at first, but I think it's better that way, because if we do it in `notifications.publish()` we'll have to handle the case where notifications are delayed after the DB transaction, so that one more layer down where we'll have to pass that flag. And since it's an issue that only affects message callbacks and is pretty specific to the way message callbacks are run, I thought it's more logical to handle these particularities there.

Reply:
Cool.